$NetBSD: patch-configure,v 1.3 2026/06/28 06:16:28 tsutsui Exp $

- Add necessary headers to test programs to avoid configure tests from
  failing unintentionally on -std=gnu23
  (mostly due to missing prototypes of exit(3) etc.)

--- configure.orig	2015-05-17 17:24:58.000000000 +0000
+++ configure
@@ -2680,6 +2680,7 @@ rm -f conftest.$ac_objext conftest.$ac_e
   cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 $ac_declaration
 #ifdef F77_DUMMY_MAIN
 #  ifdef __cplusplus
@@ -3961,6 +3962,7 @@ else
 #line $LINENO "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 # define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -4465,6 +4467,7 @@ rm -f conftest.$ac_objext conftest.$ac_e
   cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 $ac_declaration
 #ifdef F77_DUMMY_MAIN
 #  ifdef __cplusplus
@@ -18826,6 +18829,7 @@ else
 #line $LINENO "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 # define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -19230,10 +19234,13 @@ else
 #line $LINENO "configure"
 #include "confdefs.h"
 
+#if HAVE_STDLIB_H
+# include <stdlib.h>
+#endif
 #ifdef HAVE_TERMCAP_H
 # include <termcap.h>
 #endif
-main() {char *s; s=(char *)tgoto("%p1%d", 0, 1); exit(0); }
+int main() {char *s; s=(char *)tgoto("%p1%d", 0, 1); exit(0); }
 _ACEOF
 rm -f conftest$ac_exeext
 if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
@@ -19281,10 +19288,16 @@ else
 #line $LINENO "configure"
 #include "confdefs.h"
 
+#if HAVE_STRING_H
+# include <string.h>
+#endif
+#if HAVE_STDLIB_H
+# include <stdlib.h>
+#endif
 #ifdef HAVE_TERMCAP_H
 # include <termcap.h>
 #endif
-main()
+int main()
 {char *s; s=(char *)tgoto("%p1%d", 0, 1); exit(!strcmp(s==0 ? "" : s, "1")); }
 _ACEOF
 rm -f conftest$ac_exeext
@@ -19827,6 +19840,12 @@ else
   cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
+#if HAVE_STRING_H
+# include <string.h>
+#endif
+#if HAVE_STDLIB_H
+# include <stdlib.h>
+#endif
 
 #ifdef F77_DUMMY_MAIN
 #  ifdef __cplusplus
@@ -20042,6 +20061,9 @@ else
 #if HAVE_UNISTD_H
 # include <unistd.h>
 #endif
+#if HAVE_STDLIB_H
+# include <stdlib.h>
+#endif
 
 #ifdef F77_DUMMY_MAIN
 #  ifdef __cplusplus
